### DISCUSSION TITLE
Need to copy param values from json (copy text inside quotes " ")

### DISCUSSION THREAD

**Discussion by straddler** [gdh1995/vimium-c#600] [Q&A] [answered]
Hi,

I have such a workflow that I have to copy some text (parameters) from json that is display on a webpage.

Currently, I enter visual mode, then use **/** to search for start of the param (in bellow example that would be **/Value1**).
And then select everything up until end of line with **$** and then return a couple of chars back hit enter and I have it in the clipboard.
(I feel I should be able to do this better, to somehow jump to end of " but I was not able to figure that out - if anyone knows please help)

Example:
> {
> "Param1" : "/Value1INeedToCopy/with/some/path/blabla",
> "Param2" : "/Value2INeedToCopy/with/some/other/path"
> ..etc
> }

Now I am interested if somehow I can improve this? For example could I with usage of map or runKey do something better?

Ideally, how would I love this to work would be in the same manner as find (**f**) works where I would get chars I need to type to get value in-between " " to clipboard. I believe to achieve my ideal scenario would required new feature to be implemented all together? If possible I would be interested to help with implementation if that's something that is seen as useful.

P.S. Thanks for the great tool! I just started using it yDay and I already love it so much.

**Comment by gdh1995** [comment]
is the json a plain text node, or rendered in color using many div elements?



---Original---
From: ***@***.***&gt;
Date: Thu, Apr 7, 2022 18:13 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [gdh1995/vimium-c] Need to copy param values from json (copy text inside quotes " ") (Discussion #600)




 
Hi,
 
I have such a workflow that I have to copy some text (parameters) from json that is display on a webpage.
 
Currently, I enter visual mode, then use / to search for start of the param (in bellow example that would be /Value1).
 And then select everything up until end of line with $ and then return a couple of chars back hit enter and I have it in the clipboard.
 (I feel I should be able to do this better, to somehow jump to end of " but I was not able to figure that out - if anyone knows please help)
 
Example:
  
{
 "Param1" : "/Value1INeedToCopy/with/some/path/blabla",
 "Param2" : "/Value2INeedToCopy/with/some/other/path"
 ..etc
 }
  
Now I am interested if somehow I can improve this? For example could I with usage of map or runKey do something better?
 
Ideally, how would I love this to work would be in the same manner as find (f) works where I would get chars I need to type to get value in-between " " to clipboard. I believe to achieve my ideal scenario would required new feature to be implemented all together? If possible I would be interested to help with implementation if that's something that is seen as useful.
 
P.S. Thanks for the great tool! I just started using it yDay and I already love it so much.

**Reply by straddler** [reply]
It is a plain text node, everything is inside one div

**Comment by gdh1995** [comment]
The most generic way is to use `$` to go to an end and then `2h` to go left by 2 characters.

If you just want to remove the `",` and `"` tails, in a **future version (v1.98.0)** there will be a way:
1. add to `Custom key mappings` on Vimium C Options:
```
map v enterVisualMode sedKeys="1"
```
2. add to `Auto substitution of various text`:
```
1@",?$@@
```

**Comment by straddler** [comment]
Hey @gdh1995 . I saw that 1.98.0 was out and this works. Huge thanks!

I improved it a bit:

1. I have the mapping:
`map v enterVisualMode sedKeys="1"`

2. And also the regex string replace:
`1=.*"(.*?)".*=$1=`

This will replace everything before and including " and everything after and including " with the content inside " ":
Ex. asdf:"**/some/path/bla**",{asdf => **/some/path/bla** after doing the selection in visual mode & hiting enter.

And this is great. Now I want to improve on this even more but I am stuck I spent a couple of hours looking at wiki: https://github.com/gdh1995/vimium-c/wiki/Map-a-key-to-different-commands-on-different-websites and trying things out but
unable to figure out how can I chain multiple comands.
Today what I do is:
1)Do a simple search to find the location of the string I want, lets say I do / "/ and then hit enter
2)Enter visual mode with v, hit $ and hit enter
3)Success, I have content thats inside quotations in my clipboard

I see huge potential to make 2) single key however unable to understand the wiki how to do it?
Also 1) could be potentiallu squased into single key - again I am unable to understand how to use the map key :(

Could you please help? Thanks a lot!

**Comment by gdh1995** [comment, marked as answer]
The 2nd step can be replaced by such mapping rules:
```
map <v-to_lb> editText dom run="extend,forward,lineBoundary"
run xxx to_lb+autoCopy#sedKeys="1"
# then `xxx` will extend selection and copy
```

Wiki of the `run` directive: https://github.com/gdh1995/vimium-c/wiki/Auto-run-a-tree-of-commands .

BTW, the regexp is not so optimal, and can be replaced by:
```
# `matched` means to return: "replaced result" or a first "capturing group"
1="(.*?)"==,matched
```

**Reply by straddler** [reply]
Thanks a lot - this helped. I could not figure this out by my self but I think now starting to get the hang of it but still a bit blurry on the `map <v-to_lb> editText dom run="extend,forward,lineBoundary"`

If I understand it corretly, this is making that "fake" mapping (thats there for some optimization) <v-to_lb> to be used later in the run part on next line.
`"editText": "Send editing action to browser",` So editText means sending editing action to browser. 

However the `dom` part, and then `run="extend,forward,lineBoundary"` are a bit puzzling and blurry.
I am guessing it's related to what you wrote here: https://github.com/gdh1995/vimium-c/issues/114#issuecomment-581088352

**Reply by gdh1995** [reply]
`editText` itself requires `run` to learn what to do, and the content of `run` is sent to APIs like https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand and https://developer.mozilla.org/en-US/docs/Web/API/Selection/modify (`extend,xxx,xxx` will be sent to `Selection::modify`).

By default `editText` only works in an editable box, while `dom` makes `editText` try to work on normal page elements.

**Reply by straddler** [reply]
Now that explains it.
